### PR TITLE
used timestamp for created at value

### LINKED ISSUE #529: add created_at for settlement endpoint
`POST /v2/publishers/settlement` should be able to receive a `created at` value to allow for manual settlements

### PR DESCRIPTION
**Pull request by maikelmclauflin**
fixes #529

### CODE REVIEW

File: eyeshade/controllers/publishers.js [hunk 1]
```diff
@@ -59,6 +65,7 @@ v2.settlement = {
 
   validate: {
     payload: Joi.array().min(1).items(Joi.object().keys({
+      timestamp: braveJoi.date().iso().optional().description('the timestamp the settlement occurred'),
```
Comment (evq) [inline eyeshade/controllers/publishers.js]: what do you think about calling this `executed_at`?

Comment (maikelmclauflin) [reply eyeshade/controllers/publishers.js]: sure. yeah, happy to do any key

Comment (maikelmclauflin) [reply eyeshade/controllers/publishers.js]: @evq did you just want the input key to change or should it be stored on the document in that way too

Comment (evq) [reply eyeshade/controllers/publishers.js]: I think changing the field stored on the document would be good too :+1:

Comment (maikelmclauflin) [reply eyeshade/controllers/publishers.js]: would you be ok with `executedAt`? just to keep camel case consistent

Comment (evq) [reply eyeshade/controllers/publishers.js]: ah good point! sgtm :+1:

**Comment by maikelmclauflin** [comment]
ya. can do

File: test/e2e.integration.test.js [hunk 2]
```diff
@@ -236,6 +236,28 @@ test('ledger : user contribution workflow with uphold BAT wallet', async t => {
...
+FROM transactions
+WHERE
+    transaction_type = 'contribution_settlement';
+`
+  const {
+    rows
+  } = await postgres.query(select)
+
+  t.deepEqual(rows.map((entry) => _.omit(entry, ['from_account', 'to_account', 'document_id', 'id'])), [{
+    created_at: new Date(+newYear + 2000),
```
Comment (evq) [inline test/e2e.integration.test.js]: looks like `newYear` isn't defined anywhere?

Comment (maikelmclauflin) [reply test/e2e.integration.test.js]: yeah. my bad. not sure how that got through the commit. just uploaded

File: eyeshade/controllers/publishers.js [hunk 3]
```diff
@@ -149,6 +159,7 @@ module.exports.initialize = async (debug, runtime) => {
         commission: bson.Decimal128.POSITIVE_ZERO,    // conversion fee (i.e., for settlement)
         fee: bson.Decimal128.POSITIVE_ZERO,           // network fee (i.e., for settlement)
         type: '',
+        executedAt: bson.Timestamp.ZERO,              // When the settlement was executed
```
Comment (evq) [inline eyeshade/controllers/publishers.js]: should this be `bson.Date.ZERO`?

Comment (maikelmclauflin) [reply eyeshade/controllers/publishers.js]: Date input will be turned into and stored as timestamp by mongo

**Pull request merged**